### PR TITLE
[ci skip] Update documentation for Range#to_formatted_s

### PR DESCRIPTION
**Pull request by davydovanton**
Added information about default `:db` format and added information about custom formats :tada:

### CODE REVIEW

**Comment by kaspth** [comment]
I don't think we should tell people to replace and break the `RANGE_FORMATS`, but extend it. Instead your example could be `Range::RANGE_FORMATS[:example] = -> { |start, stop| ... }`.

**Comment by davydovanton** [comment]
@kaspth yes, you're right, thanks! :heart:

**Comment by davydovanton** [comment]
@kaspth fixed this

**Comment by kaspth** [comment]
I'd love it if we could make the documentation here more consistent with Date's `to_formatted_s`: https://github.com/rails/rails/blob/afc98eadb892b735a9e40fbd2a9166f15822c569/activesupport/lib/active_support/core_ext/date/conversions.rb#L28-L51 :heart:

**Comment by davydovanton** [comment]
@kaspth fixed this too :smiley:

File: activesupport/lib/active_support/core_ext/range/conversions.rb [hunk 1]
```diff
@@ -3,9 +3,25 @@ class Range
...
+  #   range.to_formatted_s       # => "1..100"
+  #   range.to_s                 # => "1..100"
+  #
+  #   range.to_formatted_s(:db)  # => "BETWEEN '1' AND '100'"
+  #   range.to_s(:db)            # => "BETWEEN '1' AND '100'"
+  #
+  # == Adding your own range formats to to_formatted_s
+  # You can add your own formats to the Range::RANGE_FORMATS hash.
+  # Use the format name as the hash key and either a Proc instance
+  # that takes a start and stop arguments as the value.
```
Comment (kaspth) [inline activesupport/lib/active_support/core_ext/range/conversions.rb]: Remove a: that takes ~~a~~ start and stop arguments as the value.

File: activesupport/lib/active_support/core_ext/range/conversions.rb [hunk 2]
```diff
@@ -3,9 +3,25 @@ class Range
...
+  #
+  #   range.to_formatted_s       # => "1..100"
+  #   range.to_s                 # => "1..100"
+  #
+  #   range.to_formatted_s(:db)  # => "BETWEEN '1' AND '100'"
+  #   range.to_s(:db)            # => "BETWEEN '1' AND '100'"
+  #
+  # == Adding your own range formats to to_formatted_s
+  # You can add your own formats to the Range::RANGE_FORMATS hash.
+  # Use the format name as the hash key and either a Proc instance
```
Comment (kaspth) [inline activesupport/lib/active_support/core_ext/range/conversions.rb]: :scissors: either

File: activesupport/lib/active_support/core_ext/range/conversions.rb [hunk 3]
```diff
@@ -3,9 +3,25 @@ class Range
...
+  # that takes a start and stop arguments as the value.
+  #
+  #   # config/initializers/range_formats.rb
+  #   Range::RANGE_FORMATS[:example] = ->(start, stop) { "you range between '#{start.to_s(:db)}' and '#{stop.to_s(:db)}'" }
```
Comment (kaspth) [inline activesupport/lib/active_support/core_ext/range/conversions.rb]: Let's use a more real example, like:

```ruby
Range::RANGE_FORMATS[:short] = ->(start, stop) { "Between #{start.to_s(:short)} and #{stop.to_s(:short)}" }
```

File: activesupport/lib/active_support/core_ext/range/conversions.rb (hunk 1 above)
Comment (davydovanton) [inline activesupport/lib/active_support/core_ext/range/conversions.rb]: entire line, yes?

**Comment by davydovanton** [comment]
@kaspth fixed all. Thanks for review! :green_heart: :yellow_heart: :heart:

**Comment by kaspth** [comment]
No problem, :+1:. The documentation here is way better because of your work :heart:

**Comment by kaspth** [comment]
Holy crap! You got the 50000th commit to Rails :wink:

**Comment by davydovanton** [comment]
WOW! :tada: thanks! :star2:

**Comment by vipulnsward** [comment]
:tada:

**Pull request merged**